### ISSUE TITLE
Sonoma County, CA Regional Parks Addition

### ISSUE THREAD

**Issue by zappy16** [juftin/camply#338]
**Is your feature request related to a problem? Please describe.**
The ability to search Sonoma County Regional Parks reservation site for campsite availability. Specifically Westside Regional Park as it's a very busy park in the summertime and weekend reservations are very difficult to get.  
https://secure.sonomacountyparks.org/camp/index.html

**Describe the solution you'd like**
Add this to the providers that Camply is able to scan

**Describe alternatives you've considered**
Manually searching the site periodically in hopes a site opens up for a weekend in the summer.

**Additional context**
The process to serach for a reservation seems to ask for your home zip code, which seems to take any value. For example a minimum of 4 numbers 1234 drops down the buttons to search by date or view by calendar. 
<img width="1143" alt="Screenshot 2024-04-25 at 9 41 51 PM" src="https://github.com/juftin/camply/assets/17417763/ef288a55-ced1-4998-9680-1164a26591af">

**Comment by zappy16** [comment]
Not sure this will work looking at the site data as best as I can tell. It seems to have a Captcha as well as render everything in the backend so you'd have to scrape the HTML. I'm not seeing any API access... wishful thinking.

**Comment by juftin** [comment]
Yeah, I looked at this as well - rendering HTML on the backend + Captcha likely makes this provider incompatible with Camply 😞
